### PR TITLE
New features for v1.0.2

### PR DESCRIPTION
**Pull request by yotamloe**
* Add logzio headers
*  Retries
* metric filtering
* comply with java 8

### CODE REVIEW

**Comment by yotamloe** [comment]
@Doron-Bargo I still need to add tests for the new features but those are the core changes

**Comment by Doron-Bargo** [comment]
can you upgrade the following:
json-path to 2.6
javax.ws.rs-api to 2.1.1

File: build.gradle [hunk 1]
```diff
@@ -20,35 +20,36 @@ apply plugin: 'java'
...
+//}
 
 signing {
     sign configurations.archives
 }
 
 dependencies {
-
-    compile 'io.micrometer:micrometer-registry-prometheus:latest.release'
+    implementation "io.github.resilience4j:resilience4j-retry:1.7.0"
```
Comment (Doron-Bargo) [inline build.gradle]: why not 1.7.1 ?

File: build.gradle [hunk 2]
```diff
@@ -20,35 +20,36 @@ apply plugin: 'java'
 apply plugin: 'org.junit.platform.gradle.plugin'
 apply plugin: 'com.google.protobuf'
 apply plugin: 'signing'
-apply plugin: 'maven-publish'
-apply plugin: 'maven'
+
+//apply plugin: 'maven-publish'
```
Comment (Doron-Bargo) [inline build.gradle]: why in comment?

File: src/test/java/io/micrometer/logzio/LogzioMeterRegistryTest.java [hunk 3]
```diff
@@ -86,6 +80,62 @@ static void destroy() {
         registry.close();
         server.stop();
     }
+//    @Test
```
Comment (Doron-Bargo) [inline src/test/java/io/micrometer/logzio/LogzioMeterRegistryTest.java]: why under comment?

**Review by Doron-Bargo — CHANGES_REQUESTED** [review]
great work
some cosmetic questions and pleas bump the relevant dependencies

File: src/test/java/io/micrometer/logzio/LogzioMeterRegistryTest.java (hunk 3 above)
Comment (yotamloe) [reply src/test/java/io/micrometer/logzio/LogzioMeterRegistryTest.java]: Still haven't created tests for the new features

**Review by Doron-Bargo — APPROVED** [review]
Great work

**Pull request merged**